src/Storages/MergeTree/PartMetadataManagerOrdinary.cpp

```diff
@@ -11,7 +11,11 @@ namespace DB
 std::unique_ptr<ReadBuffer> PartMetadataManagerOrdinary::read(const String & file_name) const
 {
     size_t file_size = part->getDataPartStorage().getFileSize(file_name);
-    auto res = part->getDataPartStorage().readFile(file_name, getReadSettings().adjustBufferSize(file_size), file_size, std::nullopt);
+    auto read_settings = getReadSettings().adjustBufferSize(file_size);
+    /// Default read method is pread_threadpool, but there is not much point in it here.
+    read_settings.local_fs_method = LocalFSReadMethod::pread;
```

Comment:
P.S. I did not do the same for remote_fs_method, because it does not use threadpool unless prefetch is made, which is not made here (while for local fs we use threadpool unconditionally, sadly this is inconsistent). Plus there is a known issue with remote_fs_method=read which is not fixed.